src/components/vue-form-renderer.vue

```diff
@@ -205,42 +207,45 @@ export default {
         this.$emit('submit', this.transientData);
       }
     },
-    validateElements(elements) {
-      elements.forEach(element => {
-        if (element.validator && element.validator.errorCount !== 0) {
-          this.valid = false;
-          this.errors.push(element.validator.errors.errors);
+    getDataAndRules(items) {
+      let data = {}, rules = {};
+      items.forEach(item => {
+
+        if (Array.isArray(item)) {
+          const [data1, rules1] = this.getDataAndRules(item);
+          data = {...data, ...data1};
+          rules = {...rules, ...rules1};
         }
-      });
-    },
-    validateContainer(container) {
-      if (container.$refs && container.$refs.container) {
-        this.validateContainer(container.$refs.container);
-      }
-      container.forEach(element => {
-        if (element.$refs && element.$refs.elements) {
-          this.validateElements(element.$refs.elements);
+
+        if (item.items) {
+          const [data2, rules2] = this.getDataAndRules(item.items);
+          data = {...data, ...data2};
+          rules = {...rules, ...rules2};
+        }
+
+        if (item.config && item.config.name && item.config.validation) {
+          data[`${item.config.name}`] = this.data[item.config.name];
+          rules[`${item.config.name}`] = item.config.validation;
         }
       });
+
+      return [data, rules];
     },
     isValid() {
       this.errors = [];
-      this.valid = true;
+      const [ data, rules ] = this.getDataAndRules(this.config);
```

Comment:
Since these values are only being passed to `Validator`, an even better name might be `getDataAndRulesForValidator`.

However, this method has two dependencies, `this.config`, and `this.data`. It would be ideal to put this into a Validator factory, and have that manage this logic. It would also make it much easier to test.

The end result would look something like this, without a direct call to `getDataAndRules`:
```js
this.dataTypeValidator = ValidatorFactory(this.config, this.data);
```